src/evrythng/utils.py

```diff
@@ -52,11 +52,13 @@ def request(request_type, resource_url, data=None, api_key=None, files=None,
 
     # Add in custom URL parameters.
     if perPage:
-        custom_query_params['perPage'] = perPage
+        query_params['perPage'] = perPage
     if pageNumber:
-        custom_query_params['page'] = pageNumber
-    if custom_query_params:
-        url += '?{}'.format(urlencode(custom_query_params))
+        query_params['page'] = pageNumber
+
+    if query_params:
+        url += '?{}'.format(urlencode(query_params))
+        url = url.replace('filter%3D', 'filter=')
```

Comment:
This has been a bit of a headache.  EVT documentation says this:

> NOTE: Obviously, the query (X) must be URL encoded for this to work, for example filter=tags=UK&tags=shipped must be encoded as filter%3Dtags%3DUK%26tags%3Dshipped.

If we do what they suggest I get filter errors.  With this in place I'm able to filter down to a single ActionType based upon its name in order to determine its EVT ID.  This filtering capability has been added to other listing methods and provides a very flexible use of query_params via request_kwargs.